src/condor_includes/condor_crypt.h

```diff
@@ -22,25 +22,52 @@
 #define CONDOR_CRYPTO
 
 #include "condor_common.h"
+#include "condor_debug.h"
 
 #include "CryptKey.h"
 
+
+class Condor_Crypto_State {
+
+public:
+    Condor_Crypto_State(Protocol proto, KeyInfo& key);
+    ~Condor_Crypto_State();
+
+    void reset();
+    const KeyInfo& getkey() { return keyInfo_; }
+
+    // the KeyInfo holds:
+    // protocol
+    // key length
+    // key data
+    // duration
+    KeyInfo       keyInfo_;
+
+    int ivec_len;
+    unsigned char *ivec;
```

Comment:
Would this be better as either a `std::unique_ptr<std::vector<unsigned char>>` or just a `std::vector<unsigned char>`?

Same for `method_key_data` and `additional`.

Can you prefix members with `m_`?